### PR TITLE
let argument checks happen in argument parsing step and not in a custom validation step

### PR DESCRIPTION
**Pull request by balikasg**
Today, in the pipelines we have a custom `validate_args` function. This, for example, does not allow as to set `default=None` and there are workarounds setting empty strings, etc. This PR tries to fix this.

### CODE REVIEW

File: python/ml4ir/base/config/parse_args.py
Comment (balikasg) [inline python/ml4ir/base/config/parse_args.py:141]: See here: https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse

File: python/ml4ir/base/pipeline.py
Comment (balikasg) [inline python/ml4ir/base/pipeline.py:195]: we should be able to have None in arguments. This blocks this. Now every check is enforced in the argparser. Another benefit is we get more detailed help messages that are dynamic (we add something to the metrics, the metrics will directly appear).

File: python/ml4ir/base/config/parse_args.py
Comment (jakemannix) [inline python/ml4ir/base/config/parse_args.py:52]: should we have "required" args, that don't allow empty string or None? 

cc: @lastmansleeping

Comment (balikasg) [reply python/ml4ir/base/config/parse_args.py:52]: setting this to `not-required` @jakemannix as in the pipeline docstring: 
```
model_file : str, optional
            Path to pretrained model file to be loaded for evaluation or retraining
```

Comment (balikasg) [reply python/ml4ir/base/config/parse_args.py:52]: @jakemannix if you have some time, please take a look at this PR.

File: python/ml4ir/base/config/parse_args.py
Comment (jakemannix) [inline python/ml4ir/base/config/parse_args.py:221]: does this require True/False or will it work with lowercase as well?

**Review by jakemannix — DISMISSED** [review]
Booleans and None are better than than int's and empty strings.  Looks good to me!

**Comment by jakemannix** [comment]
> Booleans and None are better than than int's and empty strings. Looks good to me!

but if you have any concerns @lastmansleeping , raise them before @balikasg merges ;)

We probably need checks against commandline usage in the CI... :\

File: python/ml4ir/base/config/parse_args.py
Comment (lastmansleeping) [inline python/ml4ir/base/config/parse_args.py:63]: Still worth having the comment, no? @balikasg

File: python/ml4ir/base/config/parse_args.py
Comment (lastmansleeping) [inline python/ml4ir/base/config/parse_args.py:75]: @balikasg I don't think this will work on MLS/sip-pycli.

**Review by lastmansleeping — DISMISSED** [review]
I have to block this merge until we have verified the `nargs="+"` works on MLS for single model and grid search. My guess is that the addition of spaces, will break it.

**Comment by lastmansleeping** [comment]
@balikasg Can we merge/close this?

File: python/ml4ir/base/config/parse_args.py
Comment (arvindsrikantan) [reply python/ml4ir/base/config/parse_args.py:75]: Works fine on MLS @lastmansleeping , see SearchAssistantModeling/configs/train_ner.yaml (line 13)

Comment (balikasg) [reply python/ml4ir/base/config/parse_args.py:112]: I tested with EP too.

**Review by arvindsrikantan — APPROVED** [review]
Looks good!

**Review by lastmansleeping — APPROVED** [review]
Thanks for the work @balikasg!

**Pull request merged**
